e2e/test/scenarios/question/notebook-data-source.cy.spec.ts

```diff
@@ -0,0 +1,331 @@
+      openReviewsTable();
+      openNotebook();
+      cy.findByTestId("data-step-cell").should("have.text", "Reviews").click();
+      popover().within(() => {
+        cy.findByTestId("source-database").should(
+          "have.text",
+          "Sample Database",
+        );
+        cy.findByLabelText("Reviews").should(
+          "have.attr",
+          "aria-selected",
+          "true",
+        );
+      });
+    });
+
+    it("should correctly display the source data for a simple saved question", () => {
+      visitQuestion(ORDERS_COUNT_QUESTION_ID);
+      openNotebook();
+      cy.findByTestId("data-step-cell").should("have.text", "Orders").click();
+      popover().within(() => {
+        cy.findByTestId("source-database").should(
+          "have.text",
+          "Sample Database",
+        );
+        cy.findByLabelText("Orders").should(
+          "have.attr",
+          "aria-selected",
+          "true",
+        );
+      });
+    });
+
+    it(
+      "should correctly display a table from a multi-schema database (metabase#39807)",
+      { tags: "@external" },
+      () => {
+        const dialect = "postgres";
+        const TEST_TABLE = "multi_schema";
+
+        const dbName = "Writable Postgres12";
+        const schemaName = "Wild";
+        const tableName = "Animals";
+
+        resetTestTable({ type: dialect, table: TEST_TABLE });
+        restore(`${dialect}-writable`);
+
+        cy.signInAsAdmin();
+
+        resyncDatabase({
+          dbId: WRITABLE_DB_ID,
+        });
+
+        startNewQuestion();
+        popover().within(() => {
+          cy.findByText("Raw Data").click();
+          cy.findByText(dbName).click();
+          cy.findByText(schemaName).click();
+          cy.findByText(tableName).click();
+        });
+        visualize();
+        saveQuestion("Beasts");
+
+        openNotebook();
+        cy.findByTestId("data-step-cell").should("contain", tableName).click();
+        popover().within(() => {
+          cy.findByTestId("source-database").should("have.text", dbName);
+          cy.findByTestId("source-schema").should("have.text", schemaName);
+        });
+      },
+    );
+
+    it("should correctly display a table as the model's source when editing simple model's query", () => {
+      cy.visit(`/model/${ORDERS_MODEL_ID}/query`);
+
+      cy.findByTestId("data-step-cell").should("have.text", "Orders").click();
+      popover().within(() => {
+        cy.findByTestId("source-database").should(
+          "have.text",
+          "Sample Database",
+        );
+        cy.findByLabelText("Orders").should(
+          "have.attr",
+          "aria-selected",
+          "true",
+        );
+      });
+    });
+  });
+
+  describe("saved entity as a source (aka the virtual table)", () => {
+    const modelDetails = {
+      name: "GUI Model",
+      query: { "source-table": REVIEWS_ID, limit: 1 },
+      display: "table",
+      type: "model",
+      collection_id: SECOND_COLLECTION_ID,
+    };
```

Comment:
Only `modelDetails.name` is used later on.

Reply:
Ok?
I mostly extract this for readability.

Reply:
It looks like this object was meant to be passed to `createQuestion` but instead a different (identical) object is passed there - this duplication looks like a leftover.

Reply:
Oh, could be.
Will take a look at it. Thank you.

Reply:
I was stuck at a discrepancy between how the data picker works in OSS vs EE mode so I might have missed a few other details. Thanks again.
https://github.com/metabase/metabase/issues/40223

Reply:
Addressed this comment in https://github.com/metabase/metabase/pull/40199/commits/d32bd0eda24d72d6be1e83dfb2e188a8fbe66306

Reply:
IMO the best way out is to explicitly define the type of `modelDetails`, like this: `const modelDetails: Parameters<typeof cy.createQuestion>[0] = {`. No casting and we get a proper type check.

Reply:
Sounds good.
But my only concern is what happens if we ever decide to change the order in the future?